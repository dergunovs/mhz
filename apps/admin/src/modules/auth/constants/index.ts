export const API_POST_LOGIN = 'auth/login';
export const API_GET_CHECK_AUTH = 'auth/check';

export const URL_LOGIN = '/';
