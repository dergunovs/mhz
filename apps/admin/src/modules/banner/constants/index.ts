export const URL_BANNER = '/banners';
export const URL_BANNER_CREATE = `${URL_BANNER}/create`;
export const URL_BANNER_EDIT = `${URL_BANNER}/edit`;
