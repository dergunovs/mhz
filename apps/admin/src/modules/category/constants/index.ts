export const URL_CATEGORY = '/categories';
export const URL_CATEGORY_CREATE = `${URL_CATEGORY}/create`;
export const URL_CATEGORY_EDIT = `${URL_CATEGORY}/edit`;

export const CATEGORY_FIELD_TYPE_OPTIONS = ['string', 'number', 'boolean'];
