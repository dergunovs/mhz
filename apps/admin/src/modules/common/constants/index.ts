export const URL_MAIN = '/main';
export const URL_ERROR = '/404';
