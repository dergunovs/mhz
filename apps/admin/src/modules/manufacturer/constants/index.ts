export const URL_MANUFACTURER = '/manufacturers';
