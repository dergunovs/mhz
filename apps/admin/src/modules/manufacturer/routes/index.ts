import { URL_MANUFACTURER, URL_MANUFACTURER_CREATE, URL_MANUFACTURER_EDIT } from '@/manufacturer/constants';

export const manufacturerRoutes = [
  {
    path: URL_MANUFACTURER,
    name: 'ManufacturerList',
    component: () => import('@/manufacturer/pages/ManufacturerListPage.vue'),
  },
  {
    path: URL_MANUFACTURER_CREATE,
    name: 'ManufacturerCreate',
    component: () => import('@/manufacturer/pages/ManufacturerCreatePage.vue'),
  },
  {
    path: `${URL_MANUFACTURER_EDIT}/:manufacturer`,
    name: 'ManufacturerEdit',
    component: () => import('@/manufacturer/pages/ManufacturerEditPage.vue'),
  },
];
