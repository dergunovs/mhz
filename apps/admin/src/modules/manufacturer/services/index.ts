import { useMutation, useQuery } from '@tanstack/vue-query';

import { IManufacturer } from 'mhz-types';

import { API_MANUFACTURER } from '@/manufacturer/constants';
import { api } from '@/common/services/api';

export function getManufacturers() {
  async function fn(): Promise<IManufacturer[] | undefined> {
    const { data } = await api.get(API_MANUFACTURER);

    return data;
  }

  return useQuery({
    queryKey: [API_MANUFACTURER],
    queryFn: fn,
  });
}

export function postManufacturer(options: object) {
  async function fn(formData: IManufacturer) {
    await api.post(API_MANUFACTURER, formData);
  }

  return useMutation({
    mutationKey: [API_MANUFACTURER],
    mutationFn: fn,
    ...options,
  });
}

export function deleteManufacturer(options: object) {
  async function fn(id?: number) {
    await api.delete(`${API_MANUFACTURER}/${id}`);
  }

  return useMutation({
    mutationKey: [API_MANUFACTURER],
    mutationFn: fn,
    ...options,
  });
}
