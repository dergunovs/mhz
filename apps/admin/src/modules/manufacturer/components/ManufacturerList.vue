<template>
  <UiTable
    :headers="tableHeaders"
    :isLoading="!props.manufacturers?.length"
    :modelValue="props.modelValue"
    @update:modelValue="(value) => emit('update:modelValue', value)"
    @reset="(value) => emit('reset', value)"
  >
    <template v-if="props.manufacturers?.length">
      <tr v-for="manufacturer in props.manufacturers" :key="manufacturer._id">
        <td data-grow>
          <RouterLink :to="`${URL_MANUFACTURER_EDIT}/${manufacturer._id}`">
            {{ manufacturer.title }}
          </RouterLink>
        </td>
        <td data-no-wrap>
          {{ manufacturer.country }}
        </td>
        <td data-no-wrap>
          {{ formatDate(manufacturer.dateCreated) }}
        </td>
        <td data-no-wrap>
          {{ formatDate(manufacturer.dateUpdated) }}
        </td>
        <td>
          <UiButton @click="mutate(manufacturer._id)" layout="plain">Delete</UiButton>
        </td>
      </tr>
    </template>
  </UiTable>
</template>

<script setup lang="ts">
import { useQueryClient } from '@tanstack/vue-query';

import { IManufacturer } from 'mhz-types';
import { UiTable, UiButton, toast } from 'mhz-ui';
import { formatDate, ISortOption } from 'mhz-helpers';

import { deleteManufacturer } from '@/manufacturer/services';
import { API_MANUFACTURER, URL_MANUFACTURER_EDIT } from '@/manufacturer/constants';

interface IProps {
  manufacturers?: IManufacturer[];
  modelValue: ISortOption;
}

const props = defineProps<IProps>();
const emit = defineEmits(['update:modelValue', 'reset']);

const tableHeaders = [
  { value: 'title', title: 'Manufacturer' },
  { value: 'country', title: 'Country' },
  { value: 'dateCreated', title: 'Created' },
  { value: 'dateUpdated', title: 'Updated' },
  { title: '' },
];

const queryClient = useQueryClient();

const { mutate } = deleteManufacturer({
  onSuccess: async () => {
    await queryClient.refetchQueries({ queryKey: [API_MANUFACTURER, 1], exact: true });
    toast.success('Manufacturer deleted');
  },
});
</script>
