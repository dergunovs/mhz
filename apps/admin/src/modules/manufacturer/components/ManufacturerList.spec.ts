import { describe, it, expect, beforeEach, afterEach } from 'vitest';
import { VueWrapper, enableAutoUnmount } from '@vue/test-utils';
import { dataTest } from 'mhz-helpers';

import ManufacturerList from './ManufacturerList.vue';

import { wrapperFactory } from '@/common/test';
import { URL_MANUFACTURER_EDIT } from '@/manufacturer/constants';
import { MANUFACTURERS } from '@/manufacturer/fixtures';

const manufacturerListRow = dataTest('manufacturer-list-row');
const manufacturerListLink = dataTest('manufacturer-list-link');

let wrapper: VueWrapper;

beforeEach(() => {
  wrapper = wrapperFactory(ManufacturerList, {
    props: {
      manufacturers: MANUFACTURERS.data,
    },
  });
});

enableAutoUnmount(afterEach);

describe('ManufacturerList', async () => {
  it('exists', async () => {
    expect(wrapper.findComponent(ManufacturerList)).toBeTruthy();
  });

  it('matches snapshot', async () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('shows rows if manufacturers props', async () => {
    expect(wrapper.findAll(manufacturerListRow).length).toEqual(MANUFACTURERS.data.length);

    await wrapper.setProps({ manufacturers: [] });

    expect(wrapper.find(manufacturerListRow).exists()).toBe(false);
  });

  it('generated link to manufacturer', async () => {
    const LINK = `${URL_MANUFACTURER_EDIT}/${MANUFACTURERS.data[0]._id}`;

    expect(wrapper.find(manufacturerListLink).attributes('to')).toEqual(LINK);
  });

  it('shows manufacturer content in cells', async () => {
    expect(wrapper.find(manufacturerListLink).text()).toEqual(MANUFACTURERS.data[0].title);
  });
});
