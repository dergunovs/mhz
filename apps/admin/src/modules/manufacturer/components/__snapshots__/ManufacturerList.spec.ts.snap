// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ManufacturerList > matches snapshot 1`] = `
"<table headers="[object Object],[object Object],[object Object],[object Object],[object Object]" isloading="false">
  <tbody>
    <tr data-test="manufacturer-list-row">
      <td data-grow=""><a to="/manufacturers/edit/64dd1a67f8fd2bdaa35cc735" data-test="manufacturer-list-link">Apacer</a></td>
      <td data-no-wrap="">China</td>
      <td>15</td>
      <td data-no-wrap="">Aug 16, 2023</td>
      <td data-no-wrap="">Sep 11, 2023</td>
    </tr>
    <tr data-test="manufacturer-list-row">
      <td data-grow=""><a to="/manufacturers/edit/64dd1999f8fd2bdaa35cc723" data-test="manufacturer-list-link">Palit</a></td>
      <td data-no-wrap="">China</td>
      <td>2</td>
      <td data-no-wrap="">Aug 16, 2023</td>
      <td data-no-wrap="">-</td>
    </tr>
    <tr data-test="manufacturer-list-row">
      <td data-grow=""><a to="/manufacturers/edit/64dd1907f8fd2bdaa35cc711" data-test="manufacturer-list-link">Samsung</a></td>
      <td data-no-wrap="">Korea</td>
      <td>1</td>
      <td data-no-wrap="">Aug 16, 2023</td>
      <td data-no-wrap="">-</td>
    </tr>
  </tbody>
</table>"
`;
